llama_index/vector_stores/google/generativeai/genai_extension.py

```diff
@@ -0,0 +1,534 @@
+        )
+    )
+
+
+def query_corpus(
+    *,
+    corpus_id: str,
+    query: str,
+    k: int = 4,
+    filter: Optional[Dict[str, Any]] = None,
+    client: Optional[genai.RetrieverServiceClient] = None,
+) -> List[genai.RelevantChunk]:
+    if client is None:
+        client = _default_retriever
+    response = client.query_corpus(
+        genai.QueryCorpusRequest(
+            name=str(EntityName(corpus_id=corpus_id)),
+            query=query,
+            metadata_filters=_convertFilter(filter),
+            results_count=k,
+        )
+    )
+    return list(response.relevant_chunks)
+
+
+def query_document(
+    *,
+    corpus_id: str,
+    document_id: str,
+    query: str,
+    k: int = 4,
+    filter: Optional[Dict[str, Any]] = None,
+    client: Optional[genai.RetrieverServiceClient] = None,
+) -> List[genai.RelevantChunk]:
+    if client is None:
+        client = _default_retriever
+    response = client.query_document(
+        genai.QueryDocumentRequest(
+            name=str(EntityName(corpus_id=corpus_id, document_id=document_id)),
+            query=query,
+            metadata_filters=_convertFilter(filter),
+            results_count=k,
+        )
+    )
+    return list(response.relevant_chunks)
+
+
+@dataclass
+class Passage:
+    text: str
+    ids: List[str]
+
+
+@dataclass
+class TextAnswer:
+    answer: str
+    attributed_passages: List[Passage]
+    answerable_probability: float
+
+
+def generate_text_answer(
+    *,
+    prompt: str,
+    passages: List[str],
+    answer_style: genai.AnswerStyle,
+    client: Optional[genai.TextServiceClient] = None,
+) -> TextAnswer:
+    if client is None:
+        client = _default_text_service
+    response = client.generate_text_answer(
+        genai.GenerateTextAnswerRequest(
+            question=genai.TextPrompt(text=prompt),
+            model=_default_text_service_model,
+            answer_style=answer_style,
+            grounding_source=genai.GroundingSource(
+                passages=genai.InlinePassages(
+                    passages=[
+                        genai.InlinePassage(
+                            text=chunk,
+                            id=str(uuid.uuid4()),
+                        )
+                        for chunk in passages
+                    ]
+                )
+            ),
+        )
+    )
+    return TextAnswer(
+        answer=response.answer.output,
+        attributed_passages=[
+            Passage(text=passage.text, ids=list(passage.passage_ids))
+            for passage in response.attributed_passages
+        ],
+        answerable_probability=0.9,
+    )
+
+
+def _convertToMetadata(metadata: Dict[str, Any]) -> List[genai.CustomMetadata]:
+    cs: List[genai.CustomMetadata] = []
+    for key, value in metadata.items():
+        if isinstance(value, str):
+            c = genai.CustomMetadata(key=key, string_value=value)
+        elif isinstance(value, int):
+            c = genai.CustomMetadata(key=key, int_value=value)
```

Comment:
`int_value` and `double_value` have been replaced by `numeric_value` in `CustomMetadata`.